spec.bs

```diff
@@ -2821,14 +2821,141 @@ a [=list=] of [=interest groups=] |bidIgs|, and a [=list=] of [=bid debug report
 1. Let |requestContexts| be the value of |global|'s [=associated Document's=] [=node navigable's=]
   [=traversable navigable's=] [=traversable navigable/saved Bidding and Auction request context=].
 1. If |requestContexts|[|requestId|] does not [=map/exist=], return null.
-1. Let |response| be the result of deserializing |auctionConfig|'s [=auction config/server response=]
-  according to the Bidding and Auction Services IETF standard.
+1. Set |requestContext| to |requestContexts|[|requestId|]
+1. Try to deserialize |auctionConfig|'s [=auction config/server response=] into
+   a new [=server auction response=], |response|, using |requestContext| as the
+   context, according to the Bidding and Auction Services IETF standard. If
+   deserialization is not successful, return failure.
 
     Issue: TODO: Link deserialization to IETF standard when available.
     (<a href="https://github.com/WICG/turtledove/issues/1254">WICG/turtledove#1254</a>)
-1. Construct bids based on |response|.
-1. Add the bids form the |response| to |bidIgs|.
+1. If |response|'s [=server auction response/is chaff=] field is True, return failure.
+1. If |response|'s [=server auction response/top level seller=] is not Null, return failure.
+
+   Issue: TODO: Handle Bidding and Auction Services as component auctions.
+   (<a href="https://github.com/WICG/turtledove/issues/1254">WICG/turtledove#1254</a>)
+1. Let |winningGroup| be the [=interest group=] in the [=user agent=]'s [=interest group set=]
+   with the [=interest group/owner=] equal to |response|'s
+   [=server auction response/interest group owner=] and the [=interest group/name=]
+   equal to |response|'s [=server auction response/interest group name=]. If there is no such group,
+   return failure.
+1. Let |winningAd| be the [=interest group ad=] from |winningGroup|'s [=interest group/ads=]
+   where the [=interest group ad/render url=] equals |response|'s [=server auction response/ad render url=].
+   If there is no matching ad, return failure.
+1. Let |winningAdComponents| be a new [=list=] of [=interest group ad=].
+1. [=list/For each=] |componentAd| in |response|'s [=server auction response/ad components=]:
+  1. Let |ad| be the [=interest group ad=] from |winningGroup|'s
+     [=interest group/ad components=] where the [=interest group ad/render url=]
+     equals |componentAd|. If there is no matching element, return failure.
+  1. [=list/Append=] |ad| onto |winningAdComponents|.
+1. Let |winningBid| be a new [=generated bid=] with the following [=struct/items=]:
+   : [=generated bid/id=]
+   :: (0, 0)
+   : [=generated bid/for k-anon auction=]
+   :: TODO

```

Comment:
Most of the fields were only used in component auction though, so I filled them in when I put that together (https://github.com/WICG/turtledove/commit/5e8854ecbc0541d5469274a158e2b43a952266ae).

I can move them here if you prefer.

Reply:
Both are fine to me. Just want to make sure that we're going to fill them in very soon, since that makes the spec looking not that good in the meantime.

Reply:
Yeah, I will file the pull request for that CL once this lands.